app/src/main/java/bg/zahov/app/ui/history/HistoryScreen.kt

```diff
@@ -0,0 +1,236 @@
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.fitness.app.R
+import kotlin.math.min
+
+
+@Composable
+fun HistoryScreen(historyViewModel: HistoryViewModel = viewModel(), onItemClick: (String) -> Unit) {
+    val uiState by historyViewModel.uiState.collectAsStateWithLifecycle()
+    val context = LocalContext.current
+    HistoryContent(uiState.workouts, context, onItemClick = { onItemClick(it) })
+}
+
+@Composable
+fun HistoryContent(
+    workouts: List<HistoryWorkout>,
+    context: Context,
+    onItemClick: (String) -> Unit
+) {
+    AnimatedContent(
+        workouts.size,
+        transitionSpec = {
+            fadeIn(
+                animationSpec = tween(context.resources.getInteger(R.integer.animation_duration_medium))
+            ) togetherWith fadeOut(
+                targetAlpha = context.resources.getInteger(R.integer.animation_duration_medium)
+                    .toFloat()
+            )
+        },
+        label = ""
+    ) {
+        when (it) {
+            0 -> {
+                CircularProgressIndicator(
+                    modifier = Modifier,
+                    color = MaterialTheme.colorScheme.secondary,
+                    trackColor = MaterialTheme.colorScheme.surfaceVariant
+                )
+            }
+
+            else -> {
+                LazyColumn(Modifier.fillMaxSize()) {
+                    items(items = workouts, key = { it.id }) {
+                        Workout(it) {
+                            onItemClick(it)
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Workout(item: HistoryWorkout, onItemClick: (String) -> Unit) {
+    Box(
+        Modifier
+            .fillMaxWidth()
+            .padding(4.dp)
+            .border(1.dp, colorResource(R.color.less_vibrant_text), RoundedCornerShape(4.dp))
+            .clickable { onItemClick(item.id) }
+    ) {
+        Column(
+            Modifier
+                .padding(12.dp)
+        ) {
+            Text(
+                text = item.name,
+                color = Color.White,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                style = MaterialTheme.typography.titleMedium,
+                fontWeight = FontWeight.Bold
+            )
+            Text(
+                modifier = Modifier.padding(top = 4.dp),
+                text = item.date,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                color = colorResource(R.color.less_vibrant_text),
```

Comment:
Instead of color resources you should be using the M3 themed colors.

Reply:
Question: Can we use m3 themed colors without creating a theme and applying it in main activity because our main activity currently is not rewritten?

Reply:
You can declare an M3 Compose theme and use it for Compose. You can apply the theme only for rewritten screens.

Reply:
👍